recipe/bld.bat

```diff
@@ -1,29 +1,58 @@
-cd winbuild
+set CMAKE_CONFIG=Release
 
-if %ARCH% == 32 (
-    set ARCH_STRING=x86
-) else (
-    set ARCH_STRING=x64
-)
-
-REM This is implicitly using WinSSL.  See Makefile.vc for more info.
-nmake /f Makefile.vc MODE=dll VC=%VS_MAJOR:"=% WITH_DEVEL=%LIBRARY_PREFIX% ^
-         WITH_ZLIB=dll WITH_SSH2=dll DEBUG=no ENABLE_IDN=no ENABLE_SSPI=yes ^
-         MACHINE=%ARCH_STRING% ENABLE_UNICODE=yes
+:: Create and switch to CMake build directory
+mkdir build_%CMAKE_CONFIG%
+if errorlevel 1 exit 1
+pushd build_%CMAKE_CONFIG%
 if errorlevel 1 exit 1
 
-REM This is implicitly using WinSSL.  See Makefile.vc for more info.
-nmake /f Makefile.vc mode=static VC=%VS_MAJOR:"=% WITH_DEVEL=%LIBRARY_PREFIX% ^
-         WITH_ZLIB=dll WITH_SSH2=dll DEBUG=no ENABLE_IDN=no ENABLE_SSPI=yes ^
-         MACHINE=%ARCH_STRING% ENABLE_UNICODE=yes
+:: Build and install executable and dynamic libraries
+cmake -G "Ninja" ^
+    %CMAKE_ARGS% ^
+    -D CMAKE_BUILD_TYPE:STRING="%CMAKE_CONFIG%" ^
+    -D CMAKE_INSTALL_PREFIX:PATH="%LIBRARY_PREFIX%" ^
+    -D CMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON ^
+    -D BUILD_CURL_EXE:BOOL=ON ^
+    -D BUILD_SHARED_LIBS:BOOL=ON ^
+    -D BUILD_STATIC_LIBS:BOOL=OFF ^
+    -D CURL_ZLIB:BOOL=ON ^
+    -D CURL_USE_LIBPSL:BOOL=OFF ^
+    -D CURL_USE_LIBSSH2:BOOL=ON ^
+    -D USE_WIN32_IDN:BOOL=OFF ^
+    -D CURL_USE_SCHANNEL:BOOL=ON ^
+    -D CURL_WINDOWS_SSPI:BOOL=ON ^
+    -D ENABLE_UNICODE:BOOL=ON ^
+    "%SRC_DIR%"
+if errorlevel 1 exit 1
+cmake --build . --target install --config "%CMAKE_CONFIG%" -- -v
 if errorlevel 1 exit 1
 
-REM install static library
-copy ..\builds\libcurl-vc%VS_MAJOR:"=%-%ARCH_STRING%-release-static-zlib-dll-ssh2-dll-ipv6-sspi-schannel\lib\libcurl_a.lib %LIBRARY_PREFIX%\lib\libcurl_a.lib
-if %ERRORLEVEL% GTR 3 exit 1
+:: Build and install static libraries
+cmake -G "Ninja" ^
+    %CMAKE_ARGS% ^
+    -D CMAKE_BUILD_TYPE:STRING="%CMAKE_CONFIG%" ^
+    -D CMAKE_INSTALL_PREFIX:PATH="%LIBRARY_PREFIX%" ^
+    -D CMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON ^
+    -D BUILD_CURL_EXE:BOOL=OFF ^
+    -D BUILD_SHARED_LIBS:BOOL=OFF ^
+    -D BUILD_STATIC_LIBS:BOOL=ON ^
+    -D CURL_ZLIB:BOOL=ON ^
+    -D CURL_USE_LIBPSL:BOOL=OFF ^
+    -D CURL_USE_LIBSSH2:BOOL=ON ^
+    -D USE_WIN32_IDN:BOOL=OFF ^
+    -D CURL_USE_SCHANNEL:BOOL=ON ^
+    -D CURL_WINDOWS_SSPI:BOOL=ON ^
+    -D ENABLE_UNICODE:BOOL=ON ^
+    "%SRC_DIR%"
+if errorlevel 1 exit 1
+cmake --build . --target install --config "%CMAKE_CONFIG%" -- -v
+if errorlevel 1 exit 1
 
-REM install everything else
-robocopy ..\builds\libcurl-vc%VS_MAJOR:"=%-%ARCH_STRING%-release-dll-zlib-dll-ssh2-dll-ipv6-sspi-schannel\ %LIBRARY_PREFIX% *.* /E
-if %ERRORLEVEL% GTR 3 exit 1
+:: Rename some libraries to match previously expected names
+move %LIBRARY_LIB%\libcurl.lib %LIBRARY_LIB%\libcurl_a.lib
+if errorlevel 1 exit 1
+move %LIBRARY_LIB%\libcurl_imp.lib %LIBRARY_LIB%\libcurl.lib
+if errorlevel 1 exit 1
```

Comment:
Looks like the libraries don't match the old names. So this renames them to match